c7n/resources/rds.py

```diff
@@ -274,6 +274,133 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('minor-upgrade-flag')
+class NoMinorUpgradeFlag(Filter):
+
+    schema = type_schema('minor-upgrade-flag', value={'type': 'boolean'})
+
+    def process(self, resources, event=None):
+        if self.data.get('value', True):
+            return [r for r in resources if not r['AutoMinorVersionUpgrade']]
+        else:
+            return [r for r in resources if r['AutoMinorVersionUpgrade']]
+
+
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+
+    schema = type_schema('upgrade-available')
+
+    def _list_versions(self, client, resource):
+        from distutils.version import StrictVersion as sv
+        versions = client.describe_db_engine_versions(
+            Engine=resource['Engine'],
+            EngineVersion=resource['EngineVersion'])['DBEngineVersions'][0]
+
+        version = resource['EngineVersion']
+        for v in versions['ValidUpgradeTarget']:
+            if sv(v['EngineVersion']) > sv(version):
+                version = v['EngineVersion']
+        return version
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        results = []
+        for r in resources:
+            version = self._list_versions(client, r)
+            if version != r['EngineVersion']:
+                results.append(r)
+        return results
+
+
+@filters.register('upgrade-complete')
+class UpgradeComplete(Filter):
```

Comment:
I built it from our conversation during planning... but perhaps I overthought this... I don't think it needs to be a filter at all, just do cleanup in the upgrade